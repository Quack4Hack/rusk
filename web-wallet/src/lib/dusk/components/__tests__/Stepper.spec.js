import {
	afterEach,
	describe,
	expect,
	it
} from "vitest";
import { cleanup, render } from "@testing-library/svelte";

import { Stepper } from "..";

describe("Stepper", () => {
	afterEach(cleanup);

	it("renders the Stepper component with two steps", () => {
		const { container } = render(Stepper, { props: { steps: 2, activeStep: 0 } });

		expect(container.firstChild).toMatchSnapshot();
	});

	it("renders the Stepper component with a completed step", () => {
		const { container } = render(Stepper, { props: { steps: 2, activeStep: 1 } });

		expect(container.firstChild).toMatchSnapshot();
	});

	it("renders the Stepper component with five steps", () => {
		const { container } = render(Stepper, { props: { steps: 5, activeStep: 0 } });

		expect(container.firstChild).toMatchSnapshot();
	});

	it("renders the Stepper component with five steps, with the third one being active, and the first two â€“ completed", () => {
		const { container } = render(Stepper, { props: { steps: 5, activeStep: 3 } });

		expect(container.firstChild).toMatchSnapshot();
	});
});
