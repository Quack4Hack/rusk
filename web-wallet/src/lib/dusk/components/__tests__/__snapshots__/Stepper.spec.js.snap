// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stepper > renders the Stepper component with a completed step 1`] = `
<div>
  <div
    class="dusk-stepper"
    role="tablist"
  >
    <div
      class="dusk-stepper__progress-bar"
    >
      <div
        class="dusk-stepper__progress-filler"
        style="width: 100%;"
      />
    </div>
     
    <div
      class="dusk-stepper__steps"
    >
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step dusk-stepper__step--processed"
      />
      <div
        aria-disabled="true"
        aria-selected="true"
        class="dusk-stepper__step dusk-stepper__step--processed"
      />
    </div>
  </div>
</div>
`;

exports[`Stepper > renders the Stepper component with five steps 1`] = `
<div>
  <div
    class="dusk-stepper"
    role="tablist"
  >
    <div
      class="dusk-stepper__progress-bar"
    >
      <div
        class="dusk-stepper__progress-filler"
        style="width: 0%;"
      />
    </div>
     
    <div
      class="dusk-stepper__steps"
    >
      <div
        aria-disabled="true"
        aria-selected="true"
        class="dusk-stepper__step dusk-stepper__step--processed"
      />
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step"
      />
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step"
      />
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step"
      />
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step"
      />
    </div>
  </div>
</div>
`;

exports[`Stepper > renders the Stepper component with five steps, with the third one being active, and the first two â€“ completed 1`] = `
<div>
  <div
    class="dusk-stepper"
    role="tablist"
  >
    <div
      class="dusk-stepper__progress-bar"
    >
      <div
        class="dusk-stepper__progress-filler"
        style="width: 75%;"
      />
    </div>
     
    <div
      class="dusk-stepper__steps"
    >
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step dusk-stepper__step--processed"
      />
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step dusk-stepper__step--processed"
      />
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step dusk-stepper__step--processed"
      />
      <div
        aria-disabled="true"
        aria-selected="true"
        class="dusk-stepper__step dusk-stepper__step--processed"
      />
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step"
      />
    </div>
  </div>
</div>
`;

exports[`Stepper > renders the Stepper component with two steps 1`] = `
<div>
  <div
    class="dusk-stepper"
    role="tablist"
  >
    <div
      class="dusk-stepper__progress-bar"
    >
      <div
        class="dusk-stepper__progress-filler"
        style="width: 0%;"
      />
    </div>
     
    <div
      class="dusk-stepper__steps"
    >
      <div
        aria-disabled="true"
        aria-selected="true"
        class="dusk-stepper__step dusk-stepper__step--processed"
      />
      <div
        aria-disabled="true"
        aria-selected="false"
        class="dusk-stepper__step"
      />
    </div>
  </div>
</div>
`;
